scripts/models.py

```diff
@@ -0,0 +1,244 @@
+
+
+class Contributor(BaseModel):
+    identifier: Identifier = None
+    name: str
+    email: EmailStr = None
+    url: AnyUrl = None
+    roleName: List[RoleType]
+    includeInCitation: bool = True
+    awardNumber: str = None
+
+
+class Person(Contributor):
+    affiliation: List[str]
+
+
+class Organization(Contributor):
+    contactPoint: str
+
+
+class EthicsApproval(BaseModel):
+    identifier: Union[str, AnyUrl, Identifier] = None
+    name: str
+    url: str
+
+
+class Resource(BaseModel):
+    identifier: Identifier = None
+    name: str = None
+    url: str
+    repository: Union[str, AnyUrl] = None
+    relation: Relation
+
+
+class About(BaseModel):
+    identifier: Union[str, AnyUrl, Identifier] = None
+    name: str = None
+
+
+class AccessRequirements(BaseModel):
+    status: AccessType
+    email: EmailStr = None
+    contactPoint: str = None
+    description: str = None
+    embargoedUntil: date = None
+
+
+class DandisetStat(BaseModel):
+    numberOfFiles: int = None
+    numberOfSubjects: int = None
+    numberOfSamples: int = None
+    numberOfCells: int = None
+    dataStandard: List[str] = None
+    modality: List[str] = None
+
+
+class Digest(BaseModel):
+    value: str
+    cryptoType: AnyUrl
+
+
+class BioSample(BaseModel):
+    assayType: AnyUrl
+    anatomy: AnyUrl
+
+
+class BaseDandiset(BaseModel):
+    name: str
+    description: str
+
+
+class Dandiset(BaseDandiset):
+    """A body of structured information describing a DANDI dataset
+    """
+    schemaVersion: str = Field(default="0.0.0", readonly=True)
+    identifier: Identifier = Field(readonly=True)
+
+    contributor: List[Union[Person, Organization]]
+
+    license: License
+    keywords: List[str]
+    access: List[AccessRequirements]
+    about: List[About] = None
+    study_target: List[Union[str, AnyUrl]] = None
+    protocol: List[str] = None
+    ethicsApproval: List[EthicsApproval] = None
+    relatedResource: List[Resource] = None
+    acknowledgement: str = None
+
+    # From assets
+    measurementTechnique: List[str] = Field(readonly=True)
+    variableMeasured: List[PropertyValue] = Field(readonly=True)
+    dandisetStats: DandisetStat = Field(readonly=True)
+
+
+class PublishedDandiset(Dandiset):
+    # On publish
+    version: str = Field(readonly=True)
+    datePublished: date = Field(readonly=True)
+    url: AnyUrl = Field(readonly=True)
+    contentSize: str = Field(readonly=True)
+    repository: AnyUrl = Field(readonly=True)
+    manifestLocation: Union[AnyUrl, List[AnyUrl]] = Field(readonly=True)
+    generatedBy: Optional[AnyUrl] = Field(None, readonly=True)
+
+
+class Asset(BaseModel):
+    schemaVersion: str = Field(default="0.0.0", readonly=True)
+    identifier: Identifier = Field(readonly=True)
+    contentUrl: List[AnyUrl]
+    contentSize: str
+    encodingFormat: Union[str, AnyUrl]
+    digest: Digest
+    name: str
+    datePublished: date
+    dataType: AnyUrl
+
+    path: str = None
```

Comment:
is this the path in the "dandiset semantic space"? e.g., dandiset00001/subject2/file2.nwb

Reply:
yes